engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -171,10 +171,10 @@ private enum DateGroupId {
     public static String lastBusinessDayNyOverride;
 
     /**
-     * Allows setting a custom time provider instead of actual current time. This is mainly used when setting up for a
-     * replay simulation.
+     * Allows setting a custom clock instead of actual current time. This is mainly used when setting up for a replay
+     * simulation.
      */
-    public static TimeProvider timeProvider;
+    public static Clock clock;
```

Comment:
Having a static clock here is funky, since we can support multiple replayers, where we probably want the clock to be set based upon the context of the replayer.  We are certainly no worse than we were, but we should have a ticket to address this at some point.

Reply:
I agree - it would be much better to inject and pass along a Clock instead of having a static. I'll create an issue and link it from DateTimeUtils.